lib/core/client.js

```diff
@@ -445,6 +457,31 @@ class NightwatchClient extends EventEmitter {
     return this;
   }
 
+  setNamespacedApiMethod(key, ...args) {
+    let fn;
+    let context = namespacedApi;
+    let namespace;
+
+    if (args.length === 1) {
+      fn = args[0];
+    } else if (args.length === 2) {
+      namespace = typeof args[0] == 'string' ? context[args[0]] : args[0];
+      if (namespace) {
+        context = namespace;
+      }
+
+      fn = args[1];
+    }
+
+    if (!fn) {
+      throw new Error('Method must be a declared.');
+    }
+
+    context[key] = fn;
+
+    return this;
+  }
```

Comment:
```suggestion
function setNamespacedApiMethod(key, ...args) {
  if (args.length < 1 || args.length > 2) {
    throw new Error('Invalid arguments length');
  }

  const fn = args.pop();
  const context = typeof args[0] === 'string' ? namespacedApi[args[0]] : args[0] || namespacedApi;

  if (!fn) {
    throw new Error('Method must be a declared.');
  }

  context[key] = fn;
  return this;
}
```suggestion
  setNamespacedApiMethod(key, ...args) {
    let fn;
    let context = namespacedApi;
    let namespace;

    if (args.length === 1) {
      fn = args[0];
    } else if (args.length === 2) {
      namespace = typeof args[0] == 'string' ? context[args[0]] : args[0];
      if (namespace) {
        context = namespace;
      }

      fn = args[1];
    }

    if (!fn) {
      throw new Error('Method must be a declared.');
    }

    context[key] = fn;

    return this;
  }
```
```